cmd/scan.go

```diff
@@ -24,9 +26,35 @@ import (
 	"github.com/spf13/viper"
 )
 
+// waitForAnalysisResults calls every pollFrequency seconds to the VT API and
+// checks whether an analysis is completed or not. When the analysis is completed
+// it is returned.
+func waitForAnalysisResults(cli *utils.APIClient, analysisId string, pollFrequency, timeoutLimit time.Duration) (*vt.Object, error) {
+	ticker := time.NewTicker(pollFrequency * time.Second)
+	defer ticker.Stop()
+	ctx, cancel := context.WithTimeout(context.Background(), timeoutLimit*time.Second)
+	defer cancel()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		case <-ticker.C:
+			if obj, err := cli.GetObject(vt.URL(fmt.Sprintf("analyses/%s", analysisId))); err != nil {
+				return nil, fmt.Errorf("error retrieving analysis result: %v", err)
+			} else if status, _ := obj.Get("status"); status == "completed" {
+				return obj, nil
```

Comment:
Should we return the analysis' item instead? It could be more useful

Reply:
implemented at [c9cdf7e](https://github.com/VirusTotal/vt-cli/pull/71/commits/c9cdf7e06678bd3421540014b61442f2d83e54ff)